src/hotspot/share/runtime/mutexLocker.hpp

```diff
@@ -172,7 +172,7 @@ extern Mutex* tty_lock;                          // lock to synchronize output.
 // Print all mutexes/monitors that are currently owned by a thread; called
 // by fatal error handler.
 void print_owned_locks_on_error(outputStream* st);
-
+void unlock_locks_owned_by(Thread* t);
```

Comment:
// Unlock all Mutex/Monitors currently owned by a JavaThread when executing
// `OnError` actions.
void unlock_locks_on_error(JavaThread t);

Reply:
Okay. I will remove `thread->is_Watcher_thread()`.  I keep it for `prepare_for_emergency_dump`
I think the logic fits all general Thread. Can I keep `unlock_locks_on_error(Thread* t);`

Reply:
The only time we need an unlock-all-mutexes function is for a JavaThread in the error handling code when we need to transition to native. This is not general purpose code and we shouldn't make it look like it is.